gridstatus/pjm.py

```diff
@@ -3012,8 +3013,306 @@ def get_projected_peak_tie_flow(
             ]
         ]
+
+    @support_date_range(frequency=None)
+    def get_actual_operational_statistics(
+        self,
+        date: str | pd.Timestamp,
+        end: str | pd.Timestamp | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """
+        Retrieves the actual operational statistics data from:
+        https://dataminer2.pjm.com/feed/ops_sum_prev_period/definition
+        """
+        if date == "latest":
+            return self.get_actual_operational_statistics("today")
+
+        df = self._get_pjm_json(
+            "ops_sum_prev_period",
+            start=date,
+            end=end,
+            params={
+                "fields": "datetime_beginning_utc,datetime_ending_utc,generated_at_ept,area,area_load_forecast,actual_load,dispatch_rate",
+            },
+            filter_timestamp_name="generated_at",
+            verbose=verbose,
+        )
+
+        df = df.rename(
+            columns={
+                "datetime_beginning_utc": "Interval Start",
+                "datetime_ending_utc": "Interval End",
+                "generated_at_ept": "Publish Time",
+                "area": "Area",
+                "area_load_forecast": "Area Load Forecast",
+                "actual_load": "Actual Load",
+                "dispatch_rate": "Dispatch Rate",
+            },
+        )
+
+        return (
+            df[
+                [
+                    "Interval Start",
+                    "Interval End",
+                    "Publish Time",
+                    "Area",
+                    "Area Load Forecast",
+                    "Actual Load",
+                    "Dispatch Rate",
+                ]
+            ]
+            .sort_values(["Interval Start", "Area"])
+            .reset_index(drop=True)
+        )
+
+    def _filter_active_records(
+        self,
+        df: pd.DataFrame,
+        as_of: pd.Timestamp | None,
+    ) -> pd.DataFrame:
+        """Filter out records that are terminated before the given date"""
+        df["Effective Date"] = pd.to_datetime(
+            df["Effective Date"],
+            errors="coerce",
+        ).dt.tz_localize(
+            self.default_timezone,
+        )
+
+        df["Termination Date"] = df["Termination Date"].replace(
+            "9999-12-31T00:00:00",
+            None,
+        )
+        df["Termination Date"] = pd.to_datetime(
+            df["Termination Date"],
+            errors="coerce",
+        ).dt.tz_localize(
+            self.default_timezone,
+        )
+
+        if as_of is None:
+            return df
+
+        return df[(df["Termination Date"].isna()) | (df["Termination Date"] > as_of)]
+
+    def get_pricing_nodes(
+        self,
+        as_of: str | pd.Timestamp | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """
+        Retrieves the pricing nodes data from:
+        https://dataminer2.pjm.com/feed/pnode/definition
+        """
+        if as_of == "now":
```

Comment:
Can we pass `as_of` through `utils._handle_date` to make sure it's a timestamp before it gets sent to `_filter_active_records`? I also don't know if we need to support "now" or just "today" because that's what we do in other retrieval methods.

Reply:
Also, I was testing different things in Python and discovered this unexpected result:

```python
import pandas as pd

data = pd.DataFrame(
    {"timestamp": [pd.Timestamp.now(), pd.Timestamp.now() + pd.Timedelta(minutes=10)]}
)

data['timestamp'] > 'now'

0    False
1     True
Name: timestamp, dtype: bool

data['timestamp'] > 'today'

0    False
1     True
Name: timestamp, dtype: bool
```

You can directly compare the strings "now" and "today" against a datetime column which is why passing `as_of="today"` actually already works (but I still think we should pass `as_of` through `utils._handle_date`).